### PR TITLE
Check RSA signature instead of DSA

### PR DESCRIPTION
**Pull request by richard-bt**
@prozacchiwawa @twilmoth @jjoyce-bittorrent @janfal-bt @angitbit 

Please review. The portion that I'm most leery of is the reworked `SURSAVerifier.ValidatePathWithEncodedRSASignatureAndPublicRSAKey` function. 

We've also not tested this yet, so I'm not 100% sure that it'll work - but I wanted to get some review ahead of testing so that I'm more confident of what's right/wrong when we do get to testing.

For reference:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Security/Cryptography/RSACryptoServiceProvider.cs
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Security/Cryptography/RSAPKCS1SignatureDeformatter.cs

### CODE REVIEW

File: SparkleDotNET/SUAppcastItem.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ public class SUAppcastItem : IComparable<SUAppcastItem> {
                 ItemDescription = (string)dict.ValueForKey("description");
                 FileURL = new Uri((string)enclosure.ValueForKey("url"));
                 DSASignature = (string)enclosure.ValueForKey("sparkle:dsaSignature");
+                RSASignature = (string)enclosure.ValueForKey("sparkle:rsaSignature");
```
Comment (angitbit) [inline SparkleDotNET/SUAppcastItem.cs:42]: lgtm but we'd need a new tool to gen this [using sha and private key] for appcast

Comment (richard-bt) [inline SparkleDotNET/SUAppcastItem.cs:42]: Yeah. I've branched off of your `ancast` branch in btbrowser-jenkins-helpers to add the functionality to extract the sha1 "thumbprint" from our `mini_installer.exe`.

**Comment by twilmoth** [comment]
lgtm

File: SparkleDotNET/SUAppcastItem.cs (hunk 1 above)
Comment (richard-bt) [inline SparkleDotNET/SUAppcastItem.cs:42]: @prozacchiwawa - I extracted the public key from the certificate on Windows:
http://screencast.com/t/5nAZYnSRG6I

I think the "thumbprint" section is the hash of the certificate though, not for the binary.

**Comment by richard-bt** [comment]
In talking with Art - it doesn't look like this PR is necessary/relevant. This PR just allows SparkleDotNET to perform a RSA signature check over the entire downloaded file. That's not wholly what we want, because I was thinking of working against the signature that's embedded in the file.

What we're going to do is not modify SparkleDotNET in this way. Instead we'll:
- Create a new DSA keypair on our signing server
- Have the signing server sign our binaries and provide a checksum signature along with our mini_installer.exe
- Implement checks within this/our code to check the internal signatures to verify that the mini_installer.exe that was fetched came from BitTorrent Inc.

**Pull request closed without merging**